cinema/urls.py

```diff
@@ -1,10 +1,27 @@
-from django.urls import path
+from django.urls import path, include
 
-from cinema.views import movie_list, movie_detail
+from cinema.views import (
+    MovieViewSet,
+    CinemaHallViewSet,
+    GenreList,
+    GenreDetail,
+    ActorList,
+    ActorDetail
+)
+
+from rest_framework import routers
+
+router = routers.DefaultRouter()
+router.register("movies", MovieViewSet)
+router.register("cinema_halls", CinemaHallViewSet)
 
 urlpatterns = [
-    path("movies/", movie_list, name="movie-list"),
-    path("movies/<int:pk>/", movie_detail, name="movie-detail"),
+    path("genres/", GenreList.as_view(), name="genre_list"),
+    path("genres/<int:pk>/", GenreDetail.as_view(), name="genre_detail"),
+    path("actors/", ActorList.as_view(), name="genre_list"),
+    path("actors/<int:pk>/", ActorDetail.as_view(), name="genre_detail"),
+    path("", include(router.urls))
```

Comment:
No need to include in case when you do not add basepath
Do: urlpatters = [list of urls] + router.urls